jwst/regtest/test_nirspec_fs_spec2.py

```diff
@@ -135,3 +136,25 @@ def test_pathloss_source_type(rtdata):
             if not np.allclose(slit.data, slit.pathloss_uniform, equal_nan=True):
                 bad_slits.append(idx)
     assert not bad_slits, f'Force to uniform failed for slits {bad_slits}'
+
+
+@pytest.mark.bigdata
+def test_short_calwebb_spec2_rateints(rtdata_module):
+    """Run the calwebb_spec2 pipeline on a NIRSpec Fixed-Slit _rateints exposure
+       only up to and including the flat-field step. This is a test that the
+       flat-field step does not fail processing this multi-integration input.
+    """
+    rtdata = rtdata_module
+
+    rtdata.get_data("nirspec/fs/jw01128004001_03102_00001_nrs1_new_rateints.fits")
+
+    # Run the begining of the spec2 pipeline on a (3D) _rateints file
+    args = ["calwebb_spec2", rtdata.input,
+            "--steps.pathloss.skip=True",
```

Comment:
I would let the pathloss and photom steps be applied in this test. That way we have at least one test that sends 3D data through all spec2 steps, except of course resample_spec and extract_1d, so it'll serve as a test of the 3D handling for all steps.

Reply:
Done